observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemDeploymentProcessor.java

```diff
@@ -0,0 +1,87 @@
+import org.jboss.as.server.deployment.Attachments;
+import org.jboss.as.server.deployment.DeploymentModelUtils;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessingException;
+import org.jboss.as.server.deployment.DeploymentUnitProcessor;
+import org.jboss.as.weld.WeldCapability;
+import org.wildfly.extension.micrometer.api.MicrometerCdiExtension;
+import org.wildfly.extension.micrometer.metrics.WildFlyRegistry;
+
+public class MicrometerSubsystemDeploymentProcessor implements DeploymentUnitProcessor {
+    private final boolean exposeAnySubsystem;
+    private final List<String> exposedSubsystems;
+
+    public MicrometerSubsystemDeploymentProcessor(boolean exposeAnySubsystem, List<String> exposedSubsystems) {
+        this.exposeAnySubsystem = exposeAnySubsystem;
+        this.exposedSubsystems = exposedSubsystems;
+    }
+
+    @Override
+    public void deploy(DeploymentPhaseContext deploymentPhaseContext) throws DeploymentUnitProcessingException {
+        final DeploymentUnit deploymentUnit = deploymentPhaseContext.getDeploymentUnit();
+
+        MicrometerDeploymentService.install(deploymentPhaseContext.getServiceTarget(),
+                deploymentPhaseContext,
+                deploymentUnit.getAttachment(DeploymentModelUtils.DEPLOYMENT_RESOURCE),
+                deploymentUnit.getAttachment(DeploymentModelUtils.MUTABLE_REGISTRATION_ATTACHMENT),
+                exposeAnySubsystem,
+                exposedSubsystems);
+
+        registerCdiExtension(deploymentUnit);
+    }
+
+    @Override
+    public void undeploy(DeploymentUnit context) {
+    }
+
+    private void registerCdiExtension(DeploymentUnit deploymentUnit) throws DeploymentUnitProcessingException {
+        try {
+            CapabilityServiceSupport support = deploymentUnit.getAttachment(Attachments.CAPABILITY_SERVICE_SUPPORT);
+
+
+            final WeldCapability weldCapability = support.getCapabilityRuntimeAPI(WELD_CAPABILITY_NAME, WeldCapability.class);
+            if (!weldCapability.isPartOfWeldDeployment(deploymentUnit)) {
+                // Jakarta RESTful Web Services require Jakarta Contexts and Dependency Injection. Without Jakarta
+                // Contexts and Dependency Injection, there's no integration needed
+                MICROMETER_LOGGER.noCdiDeployment();
+                return;
+            }
+            WildFlyRegistry registry = (WildFlyRegistry) deploymentUnit.getServiceRegistry()
+                    .getService(MICROMETER_REGISTRY_RUNTIME_CAPABILITY.getCapabilityServiceName()).getValue();
+
+            weldCapability.registerExtensionInstance(new MicrometerCdiExtension(registry), deploymentUnit);
+        } catch (CapabilityServiceSupport.NoSuchCapabilityException e) {
+            //We should not be here since the subsystem depends on weld capability. Just in case ...
```

Comment:
This comment isn't actually true. I don't see any capability in MicrometerSubsystemDefinition that depends on the 'org.wildfly.weld' capability. 

Is the CDI dep actually required, or could this just ignore the failure or do some info or debug logging? IOW w/o CDI I think the JVM and management API metrics would still work, and would be exported in the Micrometer format, which differs from WF Metrics format. So the subsystem could be useful even without CDI.

If CDI really is required, there needs to be a capability registered by this subsystem that requires it; otherwise the management API won't prevent anyone from building a config w/ Micrometer and w/o CDI or removing CDI from a config with Micrometer.

Reply:
You're right. CDI isn't required. Without it, the app just won't have the MeterRegistry exposed as a CDI bean, but the extension should load and run without it just fine. I've modified that logger method to log a message rather than throw an Exception.